sync/blocksync/syncer.go

```diff
@@ -78,6 +78,11 @@ func (s *BlockSyncer) Sync(ctx context.Context) error {
 	// first, check for blocks already available on disk so we don't
 	// request them from peers.
 	for blocksToFetch > 0 {
+		// Check for context cancellation before checking each block.
+		if err := ctx.Err(); err != nil {
+			return err
+		}
+
```

Comment:
this looks redundant

Reply:
`ReadBlock` is a fast database read, and the loop breaks as soon as a block isn't found, so it typically completes quickly. I added it just in case, but we could do without it.